.grit/patterns/mux_v8.md

```diff
@@ -0,0 +1,260 @@
+---
+title: Upgrade Mux SDK to v8
+---
+
+Upgrade the Mux SDK to v8
+
+tags: #js, #ts, #npm, #upgrade, #mux, #migration
+
+```grit
+engine marzano(0.1)
+language js
+
+pattern convert_config() {
+    object($properties) => $properties where {
+        $properties <: contains bubble pair($key, $value) where or {
+            $key <: `baseUrl` => js"'baseURL'",
+            $key <: `platform` => js"'fetch'" where {
+                $name = raw``,
+                $version = raw``,
+                $value <: contains bubble($name, $version) pair($key, $value) where or {
+                    $key <: `name` where {
+                        $value <: maybe string(fragment = $val) where {
+                            $name = $val
+                        }
+                    },
+                    $key <: `version` where {
+                        $value <: maybe string(fragment = $val) where  {
+                            $version = $val
+                        }
+                    }
+                },
+                $value => `(url, opts) => {
+                    let opts = opts ?? { headers: {} };
+
+                    opts.headers['x-source-platform'] = '$name | $version';
+
+                    return fetch(url, opts)
+                }`
+            }
+        },
+    }
+}
+
+pattern change_constructors() {
+    or {
+        `const { $props } = new Mux($params)` => `const mux = new Mux($params)` where and {
```

Comment:
In general we should try to keep the patterns as specific and general as possible. In this case we can drop the const.

```suggestion
        `$destruct = new Mux($params)` where {
           $destruct <: `{ $props }` => `mux`,
```